featured projects

arthurnn opened the pull request:
related to https://github.com/24pullrequests/24pullrequests/issues/952

add featured projects column and add it to the homepage.

this looks like this, when there is a feature project:
![screenshot 2015-11-27 14 32 57](https://cloud.githubusercontent.com/assets/833383/11447418/c2e27834-9513-11e5-826d-d1401368da02.png)

review @andrew

andrew commented:
Nice, can we add the avatar of the project and bump the box further up the page?

arthurnn commented:
By avatar of the project , do you mean an URL we will input on the project row? Or should we get that info from somewhere? as far as I can tell, projects dont have avatar on GitHub.

andrew commented:
Sorry, I mean the avatar of the owner on GitHub, either an org or a user

arthurnn commented:
@andrew I added the avatar of the user that submitted the project. Not sure if thats the one we wanna show, as I am not sure if that will ever show a organization avatar. If thats not it, I think we will have to add a column to the projects table to track the owner, which we can populate from the api. thoughts?

arthurnn commented:
btw, this is how it looks now
![screen shot 2015-11-27 at 3 15 31 pm](https://cloud.githubusercontent.com/assets/833383/11447759/bd73e440-9519-11e5-9f33-418d8f36b72e.png)

andrew commented:
I wasn't think of the user that submitted it, just the owner of the repo, as you say we don't currently store that, to get the avatar all we'll need is the id of the owner, alternatively we can hardcode the four featured projects this year?

arthurnn commented:
@andrew I added a avatar_url column to the project table. That will be empty for now, we can think if we want to populate that at somepoint or not. On the view we will default to @ghost if the url is nil.
Also added a method to feature a project so that will set the avatar url given the user/org id.

andrew commented:
Nice :+1:
